index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+                    status: "validation_failed"
+                schema_not_found:
+                  value:
+                    failure_reasons:
+                      - "Failed to validate against schemas: Error when trying to read from schema https://test-cdn.murmurations.network/schemas/test_schema-ver0.json: Could not read schema from HTTP, response status is 404 Not Found"
+                    status: "validation_failed"
+                missing_linked_schemas:
+                  value:
+                    failure_reasons:
+                      - "Could not read linked_schemas"
+                    status: "validation_failed"
+  /nodes:
+    post:
+      tags:
+        - Node Endpoints
+      summary: Add a node to the index
+      description: |
+        A node adds its profile to the index by posting the location of its profile (`profile_url`). The profile must include a list (`linked_schemas`) of one or more schemas against which the profile must be validated.
+
+        Each `linked_schemas` item is the name of a schema which can be found in the library using the following URL format: `{baseUrl}/schemas/{schema}.json`. For example:
+
+        `https://test-cdn.murmurations.network/schemas/test_schema-v1.json`
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostNodeRequest"
+            example:
+              profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/PostNodeResponse"
+              example:
+                data:
+                  node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+        400:
+          description: The profile_url is missing.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+              examples:
+                missing:
+                  value:
+                    message: "The profile_url parameter is missing."
+                    status: 400
+                invalid:
+                  value:
+                    message: "Invalid JSON body."
+                    status: 400
+        429:
+          $ref: "#/components/responses/TooManyRequests"
+        500:
+          $ref: "#/components/responses/ServerError"
+    get:
+      tags:
+        - Aggregator Endpoints
+      summary: Search for nodes
+      description: |
+        Aggregators can search for nodes based on any of the following parameters:
+
+        - A specific schema (`schema`)
+        - When the node was last validated by the index (`last_updated`)
```

Comment:
Shouldn't this be: "When the node was last updated (`last_updated`)" ? I assume this won't be reset when, for example, the `validate` endpoint is requested.

Reply:
Yes, that is correct. The `validate` endpoint knows nothing about the `profile_url` because the API user is sending the actual JSON profile as a payload to that endpoint, not the `profile_url`.